devtools/debug.py

```diff
@@ -108,9 +108,16 @@ def __repr__(self) -> StrType:
 class Debug:
     output_class = DebugOutput
 
-    def __init__(self, *, warnings: 'Optional[bool]' = None, highlight: 'Optional[bool]' = None):
+    def __init__(
+            self,
+            *,
+            warnings: 'Optional[bool]' = None,
+            highlight: 'Optional[bool]' = None,
+            logger_function: 'Optional[Callable[[str], None]]' = None,
```

Comment:
If `logger_function` would have the same signature as print, you could initialize it to print per default.
That way you can get rid of the conditional code in `__call__`. One might have to wrap a logger into a lambda to ignore the additonal file/flush arguments.